internal/overlord/servstate/manager.go

```diff
@@ -462,3 +463,60 @@ func (m *ServiceManager) CheckFailed(name string) {
 		}
 	}
 }
+
+var shutdownTimeout = killWait + 100*time.Millisecond
+
+// Shutdown shuts down the service manager, stopping all running services.
+func (m *ServiceManager) Shutdown() error {
+	// Determine correct dependency order for stopping services.
+	services, err := m.StopOrder(m.runningServices())
+	if err != nil {
+		return err
+	}
+	if len(services) == 0 {
+		logger.Debugf("No services to stop.")
+		return nil
+	}
+
+	// One change to stop them all.
+	logger.Noticef("Stopping all running services.")
+	st := m.state
+	st.Lock()
+	taskSet, err := Stop(st, services)
+	if err != nil {
+		st.Unlock()
+		return err
+	}
+	chg := st.NewChange("shutdown", "Shut down service manager")
+	chg.AddAll(taskSet)
```

Comment:
This looks good. Not sure on the change name/description, but the rest looks decent.

Reply:
If you have better suggestions, let me know. This seemed direct. I also used the active voice "Shut down ..." rather than "Shutting down ..." in the summary because that's what existing Pebble and snapd changes do ("Start ...", "Download ...", and so on).

Reply:
Yeah, it's always on the imperative.